lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -743,6 +744,10 @@ public function computeChangeSet(ClassMetadata $class, $entity)
 
                 $orgValue = $originalData[$propName];
 
+                if ($orgValue instanceof BackedEnum) {
+                    $orgValue = $orgValue->value;
+                }
+
                 // skip if value haven't changed
                 if ($orgValue === $actualValue) {
```

Comment:
This PR is broken, and I'm pretty sure this is why. Based on the changes in line 747, we probably also need to have:

```php
if ($actualValue instanceof BackedEnum) {
    $actualValue = $actualValue->value;
}
```

Reply:
https://github.com/doctrine/orm/issues/10249

Reply:
I don think so. Check the code a bit above:

https://github.com/HypeMC/orm/blob/a1ed32f697f66d33de42fe96b1ea90e873ba2d5f/lib/Doctrine/ORM/UnitOfWork.php#L670

then 

https://github.com/HypeMC/orm/blob/a1ed32f697f66d33de42fe96b1ea90e873ba2d5f/lib/Doctrine/ORM/UnitOfWork.php#L699

So actual value is fetched through reflection. And if you check the `ReflectionEnumProperty::getValue()`

https://github.com/HypeMC/orm/blob/a1ed32f697f66d33de42fe96b1ea90e873ba2d5f/lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php#L36-L63

It returns the `$enum->value` already. Therefore your proposed 

```php
if ($actualValue instanceof BackedEnum) {
    $actualValue = $actualValue->value;
}
```

seems redundant.

Reply:
I could be wrong. All I know is that this patch broke several of our entities that use enums, and that seemed like a possibility. I haven't tested the theory.